central/auth/service/service_impl.go

```diff
@@ -6,29 +6,51 @@ import (
 	"github.com/gogo/protobuf/types"
 	"github.com/grpc-ecosystem/grpc-gateway/runtime"
 	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/auth/datastore"
 	v1 "github.com/stackrox/rox/generated/api/v1"
-	"github.com/stackrox/rox/pkg/auth/user"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	userPkg "github.com/stackrox/rox/pkg/auth/user"
+	"github.com/stackrox/rox/pkg/errox"
 	"github.com/stackrox/rox/pkg/grpc/authn"
 	"github.com/stackrox/rox/pkg/grpc/authz"
 	"github.com/stackrox/rox/pkg/grpc/authz/allow"
 	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/sac/resources"
 	"google.golang.org/grpc"
 )
 
 var (
+	_ v1.AuthServiceServer = (*serviceImpl)(nil)
+
 	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
 		allow.Anonymous(): {
 			// GetAuthStatus does return information about the caller identity
 			// (present in context). In case it is called by an anonymous
 			// user, it will return HTTP 401 (unauthorised) which is
 			// semantically correct.
 			"/v1.AuthService/GetAuthStatus",
+			// ExchangeAuthM2MToken exchanges an identity token of a third-party
+			// OIDC provider with a Central access token, and hence needs to allow
+			// calls by anonymous users. In case no config for exchanging the token
+			// is present, it will return HTTP 401.
```

Comment:
Hm, I suppose it is a fair point.

If we decide to have a "contract definition", then I'd agree to move it to that place and have it as the source of truth instead of at two places.

Reply:
So how about changing the text here to say `it will return HTTP 4xx status code`? This most certainly will be true and we don't leak implementation details where they probably shouldn't be.